.travis/run_job.py

```diff
@@ -86,30 +77,40 @@ def should_run_sbt_project(repo, project_name, changed_paths):
 
     travis_event_type = os.environ["TRAVIS_EVENT_TYPE"]
     travis_build_stage = os.environ["TRAVIS_BUILD_STAGE_NAME"]
+    travis_commit_range = os.environ["TRAVIS_COMMIT_RANGE"]
 
     try:
         # If it's not an sbt task, we always run it no matter what.
         task = os.environ["TASK"]
     except KeyError:
-        sbt_project_name = get_project_name()
+        parser = argparse.ArgumentParser()
+        parser.add_argument("project_name", default=os.environ.get("SBT_PROJECT"))
+        parser.add_argument("--changes-in", nargs="*")
+        args = parser.parse_args()
 
-        repo = Repository(".sbt_metadata")
+        task = f"{args.project_name}-test"
 
+        change_globs = args.changes_in + [".travis.yml"] if args.changes_in else None
         if travis_event_type == "pull_request":
-            changed_paths = get_changed_paths("HEAD", "master")
+            changed_paths = get_changed_paths("HEAD", "master", globs=change_globs)
         else:
             git("fetch", "origin")
-            changed_paths = get_changed_paths(os.environ["TRAVIS_COMMIT_RANGE"])
+            changed_paths = get_changed_paths(travis_commit_range, globs=change_globs)
 
-        if should_run_sbt_project(repo, sbt_project_name, changed_paths=changed_paths):
-            task = "%s-test" % sbt_project_name
-        else:
-            print(
-                "Nothing in this patch affects %s, so skipping tests" % sbt_project_name
-            )
-            sys.exit(0)
+        sbt_repo = Repository(".sbt_metadata")
+        try:
+            if not should_run_sbt_project(sbt_repo, args.project_name, changed_paths):
+                print(
+                    f"Nothing in this patch affects {args.project_name}, so skipping tests"
+                )
+                sys.exit(0)
+        except (FileNotFoundError, KeyError):
```

Comment:
When would we throw either of these errors?

Reply:
If we're not in an SBT project, from [here](https://github.com/wellcomecollection/catalogue/blob/feature-inferrer-ci/.travis/sbt_dependency_tree.py#L49)